mm2src/kdf_walletconnect/src/error.rs

```diff
@@ -0,0 +1,111 @@
+use derive_more::Display;
+use enum_derives::EnumFromStringify;
+use relay_rpc::rpc::{self, ServiceError, SubscriptionError};
+use tokio_tungstenite_wasm::CloseFrame;
+
+/// Errors generated while parsing
+/// [`ConnectionOptions`][crate::ConnectionOptions] and creating an HTTP request
+/// for the websocket connection.
+#[derive(Debug, Display, EnumFromStringify)]
+pub enum RequestBuildError {
+    #[from_stringify("serde_qs::Error")]
+    #[display(fmt = "Failed to serialize connection query: {_0}")]
+    Query(String),
+    #[display(fmt = "Failed to add request headers")]
+    Headers,
+    #[from_stringify("url::ParseError")]
+    #[display(fmt = "Failed to parse connection URL: {_0}")]
+    Url(String),
+}
+
+#[derive(Debug, Display, EnumFromStringify)]
+pub enum ClientError {
+    ChannelClosed,
+    #[display(fmt = "RPC error: code={code} data={data:?} message={message}")]
+    Rpc {
+        code: i32,
+        message: String,
+        data: Option<String>,
+    },
+    #[display(fmt = "Invalid error response")]
+    InvalidErrorResponse,
+    #[display(fmt = "Invalid request type")]
+    InvalidRequestType,
+    #[from_stringify("serde_json::Error")]
+    SerdeError(String),
```

Comment:
if this covers also serialization errors, it would be more specific if we make two instances of these (one for serialization and one for deserialization).

Reply:
done